### PR TITLE
feat(JOML): cleanup and migrate to joml

### PR DESCRIPTION
**Pull request by pollend**
Part of https://github.com/MovingBlocks/Terasology/issues/3832

### CODE REVIEW

**Review by jdrueckert — DISMISSED** [review]

**Comment by skaldarnar** [comment]
For some reason the github diff view is totally off 🙈 Hard to review `+1,304 −1,312` when most of the time only a few imports have changed... 🙄

**Comment by pollend** [comment]
> For some reason the github diff view is totally off see_no_evil Hard to review `+1,304 −1,312` when most of the time only a few imports have changed... roll_eyes

hiding the spacing helps quite a bit. I could drop the formatting and just do it in another PR.

File: src/main/java/org/terasology/scenario/internal/systems/RegionSystem.java [hunk 1]
```diff
@@ -90,7 +90,7 @@ public void onAttackEntity(AttackEvent event, EntityRef targetEntity, org.teraso
                     } else {
                         if (!pos.equals(create.firstHit)) {
                             RegionLocationComponent loc = editedRegion.getComponent(RegionLocationComponent.class);
-                            loc.region = Region3i.createBounded(pos, create.firstHit);
+                            loc.region = new BlockRegion(pos, create.firstHit);//Region3i.createBounded(pos, create.firstHit);
```
Comment (skaldarnar) [inline src/main/java/org/terasology/scenario/internal/systems/RegionSystem.java]: I think the equivalent to `createBounded` is to build the union of those two positions.
```suggestion
                            loc.region = new BlockRegion(pos).union(create.firstHit);
```

File: src/main/java/org/terasology/scenario/internal/systems/RegionTreeSystem.java [hunk 2]
```diff
@@ -81,30 +81,35 @@
 
     @Override
     public void postBegin() {
-        Iterable<EntityRef> scenario = entityManager.getEntitiesWith(ScenarioComponent.class); // Checks for existing Scenario
+        Iterable<EntityRef> scenario = entityManager.getEntitiesWith(ScenarioComponent.class); // Checks for existing
+        // Scenario
```
Comment (skaldarnar) [inline src/main/java/org/terasology/scenario/internal/systems/RegionTreeSystem.java]:
```suggestion
		// Checks for existing scenarios
        Iterable<EntityRef> scenario = entityManager.getEntitiesWith(ScenarioComponent.class); 
```

**Review by skaldarnar — CHANGES_REQUESTED** [review]
The tip with hiding white space changes was really helpful, brought this changeset down to `+124 −132` 👍

**Review by skaldarnar — APPROVED** [review]

**Pull request merged**